src/NuGet.Services.AzureSearch/Db2AzureSearch/NewPackageRegistrationProducer.cs

```diff
@@ -215,6 +217,24 @@ private async Task<IReadOnlyList<PackageRegistrationInfo>> GetPackageRegistratio
             }
         }
 
+        private bool ShouldSkipPackageRegistration(PackageRegistration packageRegistration)
+        {
+            if (_options.Value.SkipPackagePrefixes == null)
+            {
+                return false;
+            }
+
+            foreach (var skippedPackagePrefix in _options.Value.SkipPackagePrefixes)
+            {
+                if (packageRegistration.Id.StartsWith(skippedPackagePrefix, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
```

Comment:
There was a `LogDebug` here originally, but I removed it as it added ~15 minutes to Db2AzureSearch on DEV

Reply:
o_o

Reply:
You can ask the logger if debug is enabled and skip the invocation... but yeah seems unnecessary.